src/core/resource.ts

```diff
@@ -8,10 +8,16 @@ export interface ResourceOptions {
 	pageSize?: number;
 }
 
-export type ResourceQuery = { keys: string[]; value: string | undefined };
-
-export interface Resource<S = any> {
-	(data: S[]): { resource: Resource<S>; data: S[] };
+export type ResourceQuery = { keys: string; value: string | undefined };
```

Comment:
Should this change now? it's no longer `keys`. This could just be a simple object now

Reply:
I should change the property name, I don't it being an object here because it means the user doesn't have to `Object.keys`